inc/init.php

```diff
@@ -228,6 +227,27 @@ function_exists('ob_gzhandler') &&
 mail_setup();
 
 /**
+ * Initializes the session
+ *
+ * Makes sure the passed session cookie is valid, invalid ones are ignored an a new session ID is issued
+ *
+ * @link http://stackoverflow.com/a/33024310/172068
+ */
+function init_session() {
+    global $conf;
+    session_name(DOKU_SESSION_NAME);
+    session_set_cookie_params(DOKU_SESSION_LIFETIME, DOKU_SESSION_PATH, DOKU_SESSION_DOMAIN, ($conf['securecookie'] && is_ssl()), true);
+
+    // make sure the session cookie contains a valid session ID
+    if(isset($_COOKIE[DOKU_SESSION_NAME]) && !preg_match('/^[-,a-zA-Z0-9]{1,128}$/', $_COOKIE[DOKU_SESSION_NAME])) {
```

Comment:
According to [the documentation](http://php.net/manual/en/session.configuration.php#ini.session.sid-length), at least in PHP 7.1, the session id length can be between 22 and 256 characters. So the check should at least be extended to allow 256 characters. Further, the minimum could be increased.